src/hiccup/util.clj

```diff
@@ -52,6 +52,26 @@
   String
   (to-uri [s] (URI. s)))
 
+(deftype RawString [^String s]
+  Object
+  (^String toString [this] s)
+  (^boolean equals [this other]
+    (cond
+      (instance? String other) (= s other)
```

Comment:
OK. I added this to avoid modifiyng all tests, but it is probably worth the (not too much) effort.